util/GameRuleCategories.h

```diff
@@ -0,0 +1,25 @@
+#ifndef _GameRuleCategories_h_
+#define _GameRuleCategories_h_
+
+#include "Enum.h"
+
+class GameRuleCategories
```

Comment:
does this need to be / benefit from being a class vs. a namespace?

Reply:
seemed to me the way FO_ENUM macros work it would be better to put it into class, the friend to_string method and what not, not to push it into global namespace. Plus I had some other things in mind for the class, i.e. "GetCategoryRank".

Need to double check how the macro behaves in a dedicated namespace rather than class. If "GetCategoryRank" also gets booted out of the class then might indeed not be a need for class. Doesn't make difference to me tbh so if you prefer one way over the other I do whatever fits the codebase better

Reply:
There are various uses of `FO_ENUM` not within a class... eg. `TechStatus`, `GalaxySetupOptionGeneric`, ...

That said, does it need to be an FO_ENUM? Could just do enum class if you don't use `to_string` or streaming.

Reply:
I didn't want to mess too much with internals, streaming, python side etc. so those continue to assume category is a string. I just added some overloads to "add game rule" on cpp side to accept the enum instead of string, and that does use "to_string" to convert to the "canonical" string representation. And to_string is also used to create the unordered map of category name -> rank (where rank is numerical value of the enum to keep things simple)

Reply:
I did see the FO_ENUMs used without class, but question was about separate namespace vs class, not global namespace vs class, wouldn't want to put them into global namespace I think. I assume they should work without class, within a namespace no problem, though